includes/discovery/route.inc.php

```diff
@@ -14,154 +14,327 @@
-    foreach ($vrfs_lite_cisco as $vrf_lite) {
-        $device['context_name'] = $vrf_lite['context_name'];
+//if the device does not support IP-FORWARD-MIB, we can still discover the ipv4 (only)
+//routes using RFC1213 but no way to limit the amount of routes here !!
 
-        ////////////////ipRouteDest//////////////////
-        $oid = '.1.3.6.1.2.1.4.21.1.1';
-        $resultHelp = snmp_walk($device, $oid, "-Osqn", $mib, null);
-        $resultHelp = trim($resultHelp);
-        $resultHelp = str_replace("$oid.", "", $resultHelp);
+if (! isset($ipForwardNb['0']['inetCidrRouteNumber'])) {
+    //RFC1213-MIB
+    $mib = "RFC1213-MIB";
+    $tableRoute = array();
 
-        foreach (explode("\n", $resultHelp) as $ipRouteDest) {
-            list($ip, $value) = explode(" ", $ipRouteDest);
-            $tableRoute[$ip]['ipRouteDest'] = $value;
+    $oid = '.1.3.6.1.2.1.4.21';
+    $ipRoute = snmpwalk_group($device, $oid, $mib, 1, []);
+    d_echo($res);
+    d_echo('Table routage');
+    d_echo($ipRoute);
+        echo "RFC1213 ";
+    foreach ($tableRoute as $ipRoute) {
+        if (empty($ipRoute['ipRouteDest']) || $ipRoute['ipRouteDest'] == '') {
+            continue;
         }
 
-        /////////////////ipRouteIfIndex//////////////
-        $oid = '.1.3.6.1.2.1.4.21.1.2';
-        $resultHelp = snmp_walk($device, $oid, "-Osqn", $mib, null);
-        $resultHelp = trim($resultHelp);
-        $resultHelp = str_replace("$oid.", "", $resultHelp);
-
-        foreach (explode("\n", $resultHelp) as $ipRouteIfIndex) {
-            list($ip, $value) = explode(" ", $ipRouteIfIndex);
-            $tableRoute[$ip]['ipRouteIfIndex'] = $value;
+        unset($entryClean);
+        $entryClean['inetCidrRouteDestType'] = 'ipv4';
+        $entryClean['inetCidrRouteDest'] = $ipRoute['ipRouteDest'];
+        $inetCidrRoutePfxLen = IPv4::netmask2cidr($ipRoute['ipRouteMask']); //CONVERT
+        $entryClean['inetCidrRoutePfxLen'] = $inetCidrRoutePfxLen;
+        $entryClean['inetCidrRoutePolicy'] = $ipRoute['ipRouteInfo'];
+        $entryClean['inetCidrRouteNextHopType'] = 'ipv4';
+        $entryClean['inetCidrRouteNextHop'] = $ipRoute['ipRouteNextHop'];
+        $entryClean['inetCidrRouteMetric1'] = $ipRoute['ipRouteMetric1'];
+        $entryClean['inetCidrRouteNextHopAS'] = '0';
+        $entryClean['inetCidrRouteProto'] = $ipRoute['ipRouteProto'];
+        $entryClean['inetCidrRouteType'] = $ipRoute['ipRouteType'];
+        $entryClean['inetCidrRouteIfIndex'] = $ipRoute['ipRouteIfIndex'];
+        $entryClean['context_name'] = '';
+        $entryClean['device_id'] = $device['device_id'];
+        $entryClean['updated_at'] = array('NOW()');
+        $entryClean['inetCidrRouteNextHop_device_id'] = $device['device_id'];
+        if ($entryClean['inetCidrRouteNextHop'] != '127.0.0.1' && $entryClean['inetCidrRouteNextHop'] != 'fe80::') {
+            $entryClean['inetCidrRouteNextHop_device_id'] = Device::findByIp($entryClean['inetCidrRouteNextHop'])->device_id;
```

Comment:
This has the possibility to return null, which will cause an error when you try to null->device_id

Reply:
better now.